source/features/batch-mark-files-as-viewed.tsx

```diff
@@ -18,18 +18,26 @@ function isChecked(file: HTMLElement): boolean {
 }
 
 function batchToggle(event: delegate.Event<MouseEvent, HTMLFormElement>): void {
-	if (!event.shiftKey || !previousFile) {
+	if (!event.shiftKey) {
+		return;
+	}
+
+	const isFirstToggle = previousFile === undefined;
+
+	previousFile ??= select('.js-file'); // #5484
```

Comment:
Wow I have never seen `??=` used in real life.
I had to double check its supported for the browsers we support!
https://caniuse.com/?search=%3F%3F%3D

Reply:
> Wow I have never seen `??=` used in real life.

I think it's my first time using it actually. I feel so swanky :tophat: 

> I had to double check its supported for the browsers we support!

Actually I just assumed TypeScript would transpile it